### ISSUE TITLE
Address Circular Dependency for the Import of Event

### ISSUE THREAD

**Issue by boumenot** [Azure/WALinuxAgent#1197]
See #1194 for some details.

**Comment by boumenot** [comment]
```text
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/travis/build/Azure/WALinuxAgent/tests/common/dhcp/test_dhcp.py", line 21, in <module>
    from tests.tools import *
  File "/home/travis/build/Azure/WALinuxAgent/tests/tools.py", line 31, in <module>
    import azurelinuxagent.common.event
  File "/home/travis/build/Azure/WALinuxAgent/azurelinuxagent/common/event.py", line 33, in <module>
    from azurelinuxagent.common.protocol.restapi import TelemetryEventParam, \
  File "/home/travis/build/Azure/WALinuxAgent/azurelinuxagent/common/protocol/__init__.py", line 18, in <module>
    from azurelinuxagent.common.protocol.util import get_protocol_util, \
  File "/home/travis/build/Azure/WALinuxAgent/azurelinuxagent/common/protocol/util.py", line 37, in <module>
    from azurelinuxagent.common.protocol.wire import WireProtocol
  File "/home/travis/build/Azure/WALinuxAgent/azurelinuxagent/common/protocol/wire.py", line 32, in <module>
    from azurelinuxagent.common.event import add_event, WALAEventOperation
ImportError: cannot import name add_event
```

**Comment by boumenot** [comment]
Python 3.4 stacktrace.

```text
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/imp.py", line 235, in load_module
    return load_source(name, filename, file)
  File "/home/travis/virtualenv/python3.4.6/lib/python3.4/imp.py", line 171, in load_source
    module = methods.load()
  File "<frozen importlib._bootstrap>", line 1220, in load
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/home/travis/build/Azure/WALinuxAgent/tests/common/dhcp/test_dhcp.py", line 21, in <module>
    from tests.tools import *
  File "/home/travis/build/Azure/WALinuxAgent/tests/tools.py", line 31, in <module>
    from azurelinuxagent.common.event import init_event_logger, init_event_status
  File "/home/travis/build/Azure/WALinuxAgent/azurelinuxagent/common/event.py", line 33, in <module>
    from azurelinuxagent.common.protocol.restapi import TelemetryEventParam, \
  File "/home/travis/build/Azure/WALinuxAgent/azurelinuxagent/common/protocol/__init__.py", line 18, in <module>
    from azurelinuxagent.common.protocol.util import get_protocol_util, \
  File "/home/travis/build/Azure/WALinuxAgent/azurelinuxagent/common/protocol/util.py", line 37, in <module>
    from azurelinuxagent.common.protocol.wire import WireProtocol
  File "/home/travis/build/Azure/WALinuxAgent/azurelinuxagent/common/protocol/wire.py", line 32, in <module>
    from azurelinuxagent.common.event import add_event, WALAEventOperation
ImportError: cannot import name 'add_event'
```

**Comment by pgombar** [comment]
Fixed in #1632
